### DISCUSSION TITLE
Filter search popup in Content > Pages keeps popping up in an unwelcome fashion

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3924] [Q&A] [answered]
Everytime I do anything, this dang filter section pops up and obscures the action. Incredibly annoying. 
Is there a way to make it stop?

<img width="1537" alt="Screenshot 2024-11-04 at 21 04 41" src="https://github.com/user-attachments/assets/e5f9925d-f53a-4761-882d-904893f76547">

**Comment by mworrell** [comment]
Hmmm, doesn't happen on our browsers here.

Which browser are you using?
And what are you clicking?

This panel should only show if the "Filter..." button is clicked.

**Reply by geoidesic** [reply]
It happens on every browser. I've tried Safari, Chrome and Firefox. Same behaviour on all three. Basically every time I refresh the Content > Pages page or do anything on it, like bulk delete, then the filter opens again. I.e. each visit to the Pages Overview page results in the filters being open, by default. I have to click the Filter button to close them.

**Reply by mworrell** [reply]
Does that also happen here?

https://demo.zotonic.com/en/admin/overview

**Reply by geoidesic** [reply]
No it does not.

**Reply by mworrell** [reply]
I have two theories:

1. Maybe fetch the newest version of Zotonic
2. There is some extra '#...' in the URL?

I didn't see or heard about this "automatic open" before, so am still thinking how this can happen.

**Reply by geoidesic** [reply]
No extra URL: `https://www.zotonic.localhost:8443/admin/overview`

**Reply by geoidesic** [reply]
Version is 1.0.0-rc.15

**Reply by mworrell** [reply]
We really should start a real release :-)
Can you pull the newest version?

**Reply by geoidesic** [reply]
Ok, I tried:
```git
311435170cca:/opt/zotonic$ git pull
remote: Enumerating objects: 58, done.
remote: Counting objects: 100% (58/58), done.
remote: Compressing objects: 100% (21/21), done.
remote: Total 58 (delta 42), reused 45 (delta 37), pack-reused 0 (from 0)
Unpacking objects: 100% (58/58), 117.85 KiB | 1.12 MiB/s, done.
From https://github.com/zotonic/zotonic
   618add126..f8f0f9421  master                    -> origin/master
 * [new branch]          3854-unicode-config-value -> origin/3854-unicode-config-value
 * [new branch]          category-dialog-connect   -> origin/category-dialog-connect
Updating 618add126..f8f0f9421
Fast-forward
 apps/zotonic_core/priv/translations/nl.zotonic.po                              | 4 ++--
 apps/zotonic_core/src/csv/z_csv_writer.erl                                     | 4 +++-
 apps/zotonic_launcher/bin/zotonic                                              | 4 ++--
 apps/zotonic_mod_admin/priv/templates/_admin_edit_content.query.tpl            | 2 +-
 apps/zotonic_mod_admin/priv/templates/_admin_edit_header.tpl                   | 5 +++++
 apps/zotonic_mod_admin/priv/templates/_admin_overview_filter_panel.tpl         | 4 ++--
 apps/zotonic_mod_admin/priv/templates/_admin_overview_list.tpl                 | 6 ++++++
 apps/zotonic_mod_admin/priv/templates/_admin_status_alert.tpl                  | 2 +-
 apps/zotonic_mod_admin/priv/templates/_admin_system_content_warning.tpl        | 2 +-
 apps/zotonic_mod_admin/priv/templates/admin_media.tpl                          | 7 ++++++-
 apps/zotonic_mod_mailinglist/src/controllers/controller_mailinglist_export.erl | 5 ++++-
 11 files changed, 33 insertions(+), 12 deletions(-)
 ```
 
 However, the VERSION hasn't changed, I'm not seeing the locks you added (are they in?) and the filter is still open by default

I'm seeing this in the inspector console (Chrome):
```
Error with Feature-Policy header: Unrecognized feature: 'vr'.Understand this warningAI
overview:751 Refused to apply inline style because it violates the following Content Security Policy directive: "style-src 'self' 'nonce-ZxJEo6f8spx2WIajmSqs'". Either the 'unsafe-inline' keyword, a hash ('sha256-aqNNdDLnnrDOnTNdkJpYlAxKVJtLt9CtFLklmInuUAE='), or a nonce ('nonce-...') is required to enable inline execution. Note that hashes do not apply to event handlers, style attributes and javascript: navigations unless the 'unsafe-hashes' keyword is present.
Understand this errorAI
An SSL certificate error occurred when fetching the script.Understand this errorAI
prism~zotonic-search-view~2863124690.js:3510 Could not start serviceWorker due to a SecurityError.
prism~zotonic-search-view~2863124690.js:3510 See https://cotonic.org/#model.serviceWorker for more information.
```

**Reply by mworrell** [reply]
Aha. That is your CSP header being different. It needs to allow inline css styling.

**Reply by geoidesic** [reply]
Ok I tried to create a csp module on ChatGPT's advice, to get the HMR working, then abandoned it in favour of the observer you gave me. Is it the observer or the custom csp module causing the issue?

**Reply by mworrell** [reply]
Maybe you can show me the code you are using now?
And the CSP header on the page?

**Reply by geoidesic** [reply]
Um.. I don't have a repo for that code. What files do you need to see? How do I find the CSP header on the page? (I looked at source view on the rendered page and didn't see anything like that).

**Reply by mworrell** [reply]
The CSP header you generate.  Apparently it disallowed inline css.

**Reply by geoidesic** [reply]
The only changes I've made to CSP were the ones you recommended here: https://github.com/zotonic/zotonic/discussions/3915#discussioncomment-11141467

Is that what you mean?

**Reply by mworrell** [reply]
I am curious what the actual CSP header on your request response is.  Can you copy it from your request response headers?

It seems to be a bit more restrictive that the one I imagined, maybe we have a typo somewhere.

**Reply by geoidesic** [reply]
Which request though? Assuming you mean `overview`, here it is:

<img width="1192" alt="Screenshot 2024-11-26 at 12 58 10" src="https://github.com/user-attachments/assets/b5d5f775-b828-41ed-90c4-d0d8b9439760">

**Reply by mworrell** [reply]
Yes, and could you copy the text of the CSP header in the "Response"?

**Reply by geoidesic** [reply]
```=
Response
:status: 200
Cache-Control: no-store, no-cache, must-revalidate, private, post-check=0, pre-check=0
Content-Length: 74084
Content-Security-Policy: default-src 'self'; object-src 'none'; style-src 'self' 'nonce-W5Pl9vTXjfNmjxRuydRi'; img-src 'self' data:; media-src 'self' data:; font-src 'self'; frame-src 'self'; form-action 'self'; worker-src 'self'; connect-src 'self' wss://localhost:5173; script-src 'self' 'nonce-W5Pl9vTXjfNmjxRuydRi' https://localhost:5173;
Content-Type: text/html;charset=utf-8
Date: Tue, 26 Nov 2024 10:57:34 GMT
expect-ct: max-age=86400, enforce
Expires: Wed, 10 Dec 2008 14:30:00 GMT
feature-policy: autoplay 'none';camera 'none';fullscreen 'self';geolocation 'none';microphone 'none';midi 'none';payment 'none';vr 'none';
p3p: CP="NOI ADM DEV PSAi COM NAV OUR OTRo STP IND DEM"
Pragma: nocache
Referrer-Policy: no-referrer
Server: Zotonic
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
x-permitted-cross-domain-policies: none
x-robots-tag: noindex,nofollow
```

**Reply by mworrell** [reply, marked as answer]
Can you add:

```
style-src-attr 'self' 'unsafe-inline';
```

To this list?

**Reply by geoidesic** [reply]
Done:
```
default-src 'self'; object-src 'none'; style-src 'self' 'nonce-HtW7ZgHrCDnsMrR1uoaS'; img-src 'self' data:; media-src 'self' data:; font-src 'self'; frame-src 'self'; form-action 'self'; worker-src 'self'; connect-src 'self' wss://localhost:5173; script-src 'self' 'nonce-HtW7ZgHrCDnsMrR1uoaS' https://localhost:5173;
```
<img width="1192" alt="Screenshot 2024-11-28 at 12 59 46" src="https://github.com/user-attachments/assets/087e36c6-cdc4-48ad-832a-72adb8c69ad6">

That hasn't helped.

**Reply by mworrell** [reply]
I don't see the `style-src-attr` in that list?

**Reply by geoidesic** [reply]
I did add it in.. and then I cleared cache and the above was the result.

<img width="1136" alt="Screenshot 2024-11-28 at 14 06 40" src="https://github.com/user-attachments/assets/d5bef8fb-1529-4078-ba51-38aab335d91e">

**Reply by mworrell** [reply]
Make sure all is compiled and loaded. Your code looks ok to me.

BTW the script src - why are the urls for `wss://....` and `https:/...` there?   Do you connect to a separate service other than the site (`self`) ?

**Reply by geoidesic** [reply]
It was a typo... apparently erlang requires a space between ; and ".

Now it is working and the filters are not misbehaving!

<img width="1624" alt="Screenshot 2024-11-28 at 14 14 36" src="https://github.com/user-attachments/assets/96416a68-53df-456f-8a52-ee25c64a0160">

**Reply by geoidesic** [reply]
> BTW the script src - why are the urls for `wss://....` and `https:/...` there? Do you connect to a separate service other than the site (`self`) ?

I don't know? What do you mean? I'm not seeing that anywhere? No this is basically the demo site provided by Zotonic. I've only removed some of the JavaScript and CSS includes, I haven't added anything.

**Reply by geoidesic** [reply]
Oh I see it now, in the config. Probably that's a copy / pasta error.
